pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+        if property_id[prop.attrib['id']] not in [
+            'temperature', 'pressure', 'ignition-delay', 'pressure-rise'
+            ]:
+            raise KeyError(property_id[prop.attrib['id']] + ' not valid dataPoint property')
+    if property_id == {}:
+        raise AttributeError('Error: at least one property needed in dataGroup.')
+
+    # now get data points
+    datapoints = []
+    for dp in dataGroup.findall('dataPoint'):
+        datapoint = {}
+        for val in dp:
+            units = unit_id[val.tag]
+            if units == 'Torr':
+                units = 'torr'
+            datapoint[property_id[val.tag]] = [val.text + ' ' + units]
+        datapoints.append(datapoint)
+
+    if len(datapoints) == 0:
+        raise AttributeError('Error: need at least one dataPoint.')
+
+    # RCM files may have a second dataGroup with volume-time history
+    if len(dataGroups) == 2:
+        assert root.find('apparatus/kind').text == 'rapid compression machine', \
+               'Second dataGroup only valid for RCM.'
+
+        assert len(datapoints) == 1, 'Multiple datapoints for single volume history.'
+
+        dataGroup = dataGroups[1]
+        for prop in dataGroup.findall('property'):
+            if prop.attrib['name'] == 'time':
+                time_dict = {'units': prop.attrib['units'], 'column': 0}
+                time_tag = prop.attrib['id']
+            elif prop.attrib['name'] == 'volume':
+                volume_dict = {'units': prop.attrib['units'], 'column': 1}
+                volume_tag = prop.attrib['id']
+
+        volume_history = {'time': time_dict, 'volume': volume_dict, 'values': []}
+
+        # collect volume-time history
+        for dp in dataGroup.findall('dataPoint'):
+            time = None
+            volume = None
+            for val in dp:
+                if val.tag == time_tag:
+                    time = float(val.text)
+                elif val.tag == volume_tag:
+                    volume = float(val.text)
+            volume_history['values'].append([time, volume])
+
+        datapoints[0]['volume-history'] = volume_history
+
+    elif len(dataGroups) > 2:
+        raise NotImplementedError('More than two DataGroups not supported.')
+
+    return datapoints
+
+
+def convert_ReSpecTh(filename_xml, output='', file_author='', file_author_orcid=''):
+    """Convert ReSpecTh XML file to ChemKED YAML file.
+
+    Args:
+        filename_xml (`str`): Name of ReSpecTh XML file to be converted.
+        output (`str`, optional): Output path for converted file.
+        file_author (`str`, optional): Name to override original file author
+        file_author_orcid (`str`, optional): ORCID of file author
+
+    Returns:
+        filename_yaml (`str`): Name of newly created ChemKED YAML file.
+    """
+    assert os.path.isfile(filename_xml), 'Error: ' + filename_xml + ' file missing'
+
+    # get all information from XML file
+    try:
+        tree = etree.parse(filename_xml)
+    except OSError:
```

Comment:
Why is this `try` necessary? Is the `OSError` default message not clear? (I'm really not sure what the default message looks like :smile:)